wagtail/admin/views/pages.py

```diff
@@ -1109,3 +1111,36 @@ def revisions_compare(request, page_id, revision_id_a, revision_id_b):
         'revision_b': revision_b,
         'comparison': comparison,
     })
+
+
+def revisions_unschedule(request, page_id, revision_id):
+    page = get_object_or_404(Page, id=page_id).specific
+
+    user_perms = UserPagePermissionsProxy(request.user)
+    if not user_perms.for_page(page).can_unpublish():
+        raise PermissionDenied
+
+    revision = get_object_or_404(page.revisions, id=revision_id)
```

Comment:
Should check here if this revision is actually a future scheduled revision.
If not - it should redirect the user back to the revisions page, maybe with a warning that there is no revision to unschedule.